WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -2,20 +2,20 @@
 
 module CompetitionsHelper
   def competition_message_for_user(competition, user)
-    messages = []
-    registration = competition.registrations.find_by_user_id(user.id)
-    if competition.cancelled?
-      messages << t('competitions.messages.cancelled')
-    elsif registration
-      messages << (registration.accepted? ? t('competitions.messages.tooltip_registered') : t('competitions.messages.tooltip_waiting_list'))
+    # Generates a list of messages, which will be combined and displayed in a tooltip to the user in their bookmarked
+    # competitions list when they hover over a competition.
+    # Message indicates the state of the competition, and the state of the user's registration.
+
+    messages_to_join = []
```

Comment:
Super nitpick, not a blocker but more a guidance: Don't name variables according to what will be done with them later. `messages` is totally fine, and calling it `messages_to_join` is considered bad style because it "forbids" you to do anything else with that variable except joining.